hail/python/hail/ir/base_ir.py

```diff
@@ -68,6 +68,9 @@ def _eq(self, other):
         """
         return True
 
+    def __hash__(self):
+        return 31 + hash(str(self))
```

Comment:
Why add 31 here?

Reply:
This seems like it should be a `*`?

Reply:
shouldn't really matter -- I just wanted the hash of the IR to be different from the hash of the str.